src/storage/mvcc/reader/reader.rs

```diff
@@ -1066,8 +1067,8 @@ pub mod tests {
         engine.prewrite(m.clone(), k, 1);
         engine.commit(k, 1, 5);
 
-        engine.rollback(k, 3);
-        engine.rollback(k, 7);
+        engine.rollback(k, 3, false);
+        engine.rollback(k, 7, false);
```

Comment:
Why must we not protect these rollbacks? It seems odd, given we never do this in non-test code

Reply:
It's just because I found it hard to fix this test 😂.

I think this test is used to test the behavoir of `reader` and the `engine.rollback`s are used only to provide a testing environment. i.e. They are not supposed to really do the "rollback" work here. But a large part of this test depends on the fact that these rollbacks are not protected.

Moreover, who can promise that the "rollback" or "cleanup" function we are using here is right? Maybe the tests for "cleanup", but "cleanup"'s correctness is depend on the correctness of `reader`, and lead us to circular reasoning. 

Just as I said, it might be a bad habit to prepare **unit** testing environment for lower level componets (scanner in this case) with higher level components (txn and actions, or more specific, `cleanup` in this case), after all lower level componets should not know higher level components' existance. But **huge** amount of tests in TiKV are written in this way, and many of them are too hard to change.

As for the tests here, I'll try to write several `Modify`s directly into the engine here.